src/global/pid_lut/pid_lut.cc

```diff
@@ -0,0 +1,247 @@
+                         8.0, 8.2, 8.4, 8.6, 8.8, 9.0, 9.2, 9.4, 9.6, 9.8, 10.0, 10.2},
+      .polar_edges    = {25.0,  26.0,  27.0,  28.0,  29.0,  30.0,  31.0,  32.0,  33.0,  34.0,  35.0,
+                         36.0,  37.0,  38.0,  39.0,  40.0,  41.0,  42.0,  43.0,  44.0,  45.0,  46.0,
+                         47.0,  48.0,  49.0,  50.0,  51.0,  52.0,  53.0,  54.0,  55.0,  56.0,  57.0,
+                         58.0,  59.0,  60.0,  61.0,  62.0,  63.0,  64.0,  65.0,  66.0,  67.0,  68.0,
+                         69.0,  70.0,  71.0,  72.0,  73.0,  74.0,  75.0,  76.0,  77.0,  78.0,  79.0,
+                         80.0,  81.0,  82.0,  83.0,  84.0,  85.0,  86.0,  87.0,  88.0,  89.0,  90.0,
+                         91.0,  92.0,  93.0,  94.0,  95.0,  96.0,  97.0,  98.0,  99.0,  100.0, 101.0,
+                         102.0, 103.0, 104.0, 105.0, 106.0, 107.0, 108.0, 109.0, 110.0, 111.0, 112.0,
+                         113.0, 114.0, 115.0, 116.0, 117.0, 118.0, 119.0, 120.0, 121.0, 122.0, 123.0,
+                         124.0, 125.0, 126.0, 127.0, 128.0, 129.0, 130.0, 131.0, 132.0, 133.0, 134.0,
+                         135.0, 136.0, 137.0, 138.0, 139.0, 140.0, 141.0, 142.0, 143.0, 144.0, 145.0,
+                         146.0, 147.0, 148.0, 149.0, 150.0, 151.0, 152.0, 153.0, 154.0, 155.0, 156.0,
+                         157.0, 158.0, 159.0, 160.0},
+      .azimuthal_binning = {0.0, 30.5, 0.5}, // lower, upper, step
+  };
+
+  app->Add(new JOmniFactoryGeneratorT<PIDLookup_factory>(
+      "DIRCTruthSeededLUTPID",
+      {
+          "ReconstructedTruthSeededChargedWithPFRICHTOFPIDParticles",
+          "ReconstructedTruthSeededChargedWithPFRICHTOFPIDParticleAssociations",
+      },
+      {
+          "ReconstructedTruthSeededChargedWithPFRICHTOFDIRCPIDParticles",
+          "ReconstructedTruthSeededChargedWithPFRICHTOFDIRCPIDParticleAssociations",
+          "DIRCTruthSeededParticleIDs",
+      },
+      dirc_pid_cfg, app));
+
+  app->Add(new JOmniFactoryGeneratorT<PIDLookup_factory>(
+      "DIRCLUTPID",
+      {
+          "ReconstructedChargedWithPFRICHTOFPIDParticles",
+          "ReconstructedChargedWithPFRICHTOFPIDParticleAssociations",
+      },
+      {
+          "ReconstructedChargedWithPFRICHTOFDIRCPIDParticles",
+          "ReconstructedChargedWithPFRICHTOFDIRCPIDParticleAssociations",
+          "DIRCParticleIDs",
+      },
+      dirc_pid_cfg, app));
+
+  //-------------------------------------------------------------------------
+  // DRICH PID
+  //-------------------------------------------------------------------------
+
+  int ForwardRICH_ID = 0;
+  try {
+    auto detector  = app->GetService<DD4hep_service>()->detector();
+    ForwardRICH_ID = detector->constant<int>("ForwardRICH_ID");
+  } catch (const std::runtime_error&) {
+    // Nothing
+  }
+  PIDLookupConfig drich_pid_cfg{
+      .filename                 = "calibrations/drich.lut",
+      .system                   = ForwardRICH_ID,
```

Comment:
(OT) Not for this PR, but at some point I would like to remove these constructs (which require plugins to be loaded in a certain order that we are unable to enforce) and move it into algorithm initialization:
```suggestion
  PIDLookupConfig drich_pid_cfg{
      .filename                 = "calibrations/drich.lut",
      .system                   = "ForwardRICH_ID",
```
Doesn't work in all cases, but it does work in several of them.

Reply:
I'll try and do this next.